src/pages/Pool/PositionPage.tsx

```diff
@@ -609,20 +758,70 @@ export function PositionPage() {
             onDismiss={() => setShowConfirm(false)}
             attemptingTxn={collecting}
             hash={collectMigrationHash ?? ''}
-            content={() => (
-              <ConfirmationModalContent
-                title="Claim fees"
-                onDismiss={() => setShowConfirm(false)}
-                topContent={modalHeader}
-              />
-            )}
-            pendingText="Collecting fees"
+            content={() =>
+              showVioletEmbed && isRegistered && authorizeProps ? (
+                violetError ? (
+                  <TransactionErrorContent
+                    onDismiss={handleDismissConfirmation}
+                    message={handleErrorCodes(violetError)}
+                  />
+                ) : (
+                  <EmbeddedAuthWrapper>
+                    <EmbeddedAuthorization
+                      ref={embeddedAuthRef}
+                      authorizeProps={authorizeProps}
+                      onIssued={({ signature, expiry }: any) => {
+                        if (!call) {
+                          throw new Error('Missing call following EAT issuance')
+                        }
+                        handleVioletResponseAndSubmitTransaction({
+                          ...signature,
+                          expiry,
+                          to: call.address,
+                          calls: call.calls,
+                          value: call.value,
+                        })
+                      }}
+                      onFailed={(response: any) => {
+                        console.error(`Violet Embedded Auth failed: ${JSON.stringify(response, null, 2)}`)
+                        setVioletError(JSON.stringify(response?.code))
+                      }}
+                    />
+                  </EmbeddedAuthWrapper>
+                )
+              ) : violetError ? (
+                <TransactionErrorContent
+                  onDismiss={handleDismissConfirmation}
+                  message={handleErrorCodes(violetError)}
+                />
+              ) : pendingVioletAuth ? (
+                <ConfirmationPendingContent onDismiss={handleDismissConfirmation} pendingText={pendingText} />
+              ) : (
+                <ConfirmationModalContent
+                  title="Claim fees"
+                  onDismiss={handleDismissConfirmation}
+                  topContent={modalHeader}
+                />
+              )
+            }
```

Comment:
There are so many ternaries chained that it's hard to read

Reply:
I didn't change the implementation from what was there before, to not cause too many changes

Reply:
Ok